db/fixtures/ae_datastore/ManageIQ/ConfigurationManagement/AnsibleTower/Operations/StateMachines/Job.class/__methods__/wait_for_completion.rb

```diff
@@ -0,0 +1,48 @@
+#
+# Description: Given a Ansible Job Id, check it's status
+#
+class WaitForCompletion
+  JOB_CLASS = 'ManageIQ_Providers_AnsibleTower_ConfigurationManager_Job'.freeze
+  def initialize(handle)
+    @handle = handle
+    @job_id = @handle.get_state_var(:ansible_job_id)
+    if @job_id.nil?
+      @handle.log(:error, 'Ansible job id not found')
+      exit(MIQ_ERROR)
```

Comment:
Instead of exiting from the `initialize` method I think it would be better to move all the job logic into a `ansible_job` method that can validate both the job_id and job object when called from `main`.

Something like:

```ruby
def main
  check_status(ansible_job)
end
```